core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -233,19 +235,21 @@ public void updateDataMapSchema(List<DataMapSchema> dataMapSchemaList,
    * @param dataMapProvider
    * @param dataMapSchema
    */
-  public synchronized void registerDataMapCatalog(DataMapProvider dataMapProvider,
+  public void registerDataMapCatalog(DataMapProvider dataMapProvider,
       DataMapSchema dataMapSchema) throws IOException {
-    initializeDataMapCatalogs(dataMapProvider);
-    String name = dataMapSchema.getProviderName();
-    DataMapCatalog dataMapCatalog = dataMapCatalogs.get(name);
-    if (dataMapCatalog == null) {
-      dataMapCatalog = dataMapProvider.createDataMapCatalog();
-      if (dataMapCatalog != null) {
-        dataMapCatalogs.put(name.toLowerCase(), dataMapCatalog);
+    synchronized (lockObject) {
+      initializeDataMapCatalogs(dataMapProvider);
```

Comment:
In initializeDataMapCatalogs,  dataMapCatalogs.get(schema.getProviderName()), here provider name should be converted to lowercase, because put is happened with lowercase, else look up will return null..

Reply:
done